pkg/build/builder/cmd/scmauth/util_test.go

```diff
@@ -0,0 +1,38 @@
+package scmauth
+
+import (
+	buildv1 "github.com/openshift/api/build/v1"
+	"os"
+	"testing"
+)
+
+func TestEnsureGitConfigIncludesProxyConfig(t *testing.T) {
```

Comment:
I think the following scenarios need to be tested:

1. Proxy variables are not defined
2. `GIT_CONFIG` is not set (verify error returned)
3. Permutations of proxy variables:
   1. HTTPS_PROXY only set
   2. HTTP_PROXY only set
   3. Both set

Reply:
additional test case scenarios pushed @adambkaplan